gen/cpp/sym/factors/internal_imu_factor.h

```diff
@@ -0,0 +1,2373 @@
+// -----------------------------------------------------------------------------
+// This file was autogenerated by symforce from template:
+//     function/FUNCTION.h.jinja
+// Do NOT modify by hand.
+// -----------------------------------------------------------------------------
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/pose3.h>
+#include <sym/rot3.h>
+
+namespace sym {
+
+/**
+ * An internal helper function to linearize the difference between the orientation, velocity, and
+ * position at one time step and those of another time step. The expected difference is calculated
+ * from the preintegrated IMU measurements between those time steps.
+ *
+ * NOTE: If you are looking for a residual for an IMU factor, do not use this. Instead use
+ * the one found in symforce/slam/imu_preintegration/imu_factor.h.
+ *
+ * Time step i is the time of the first IMU measurement of the interval.
+ * Time step j is the time after the last IMU measurement of the interval.
+ *
+ * Assumes sqrt_info is lower triangular and only reads the lower triangle.
+ *
+ * Args:
+ *     pose_i (sf.Pose3): Pose at time step i (maps body frame at time i to world frame)
+ *     vel_i (sf.V3): Velocity at time step i (world frame)
+ *     pose_j (sf.Pose3): Pose at time step j (maps body frame at time j to world frame)
+ *     vel_j (sf.V3): Velocity at time step j (world frame)
+ *     accel_bias_i (sf.V3): The accelerometer bias between timesteps i and j
+ *     gyro_bias_i (sf.V3): The gyroscope bias between timesteps i and j
+ *     DR (sf.Rot3): Preintegrated estimate for pose_i.R.inverse() * pose_j.R
+ *     Dv (sf.V3): Preintegrated estimate for vel_j - vel_i in the body frame at timestep i
+ *     Dp (sf.V3): Preintegrated estimate for pos change (before velocity and gravity corrections)
+ *         in the body frame at timestep i: R_i^T (p_j - p_i - v_i Delta t - 1/2 g Delta t^2),
+ *         where R_i = pose_i.R, p_i = pose_i.t, p_j = pose_j.t, and v_i = vel_i
+ *     sqrt_info (sf.M99): sqrt info matrix of DR('s tangent space), Dv, Dp
+ *     DR_D_gyro_bias (sf.M33): Derivative of DR w.r.t. gyro_bias evaluated at gyro_bias_hat
+ *     Dp_D_accel_bias (sf.M33): Derivative of Dp w.r.t. accel_bias evaluated at accel_bias_hat
+ *     Dp_D_gyro_bias (sf.M33): Derivative of Dp w.r.t. gyro_bias evaluated at gyro_bias_hat
+ *     Dv_D_accel_bias (sf.M33): Derivative of Dv w.r.t. accel_bias evaluated at accel_bias_hat
+ *     Dv_D_gyro_bias (sf.M33): Derivative of Dv w.r.t. gyro_bias evaluated at gyro_bias_hat
+ *     accel_bias_hat (sf.V3): The accelerometer bias used during preintegration
+ *     gyro_bias_hat (sf.V3): The gyroscope bias used during preintegration
+ *     gravity (sf.V3): Acceleration due to gravity (in the same frame as pose_x and vel_x),
+ *         i.e., the vector which when subtracted from the accelerometer measurements gives the
+ *         true acceleration (up to bias and noise) of the IMU.
```

Comment:
I think instead of "true acceleration" I might say "acceleration relative to the world frame" or something

Reply:
Well, the distinction I was trying to make was between the sort of hallucinated acceleration IMUs have upwards and the actual acceleration in the real world.

I also worry mentioning frames might confuse matters more, as the gravity vector is expected to be written in the world frame, but the acceleration measurements are in the body frame. So, before you add their coordinates you have to convert one or the other.

As an aside, I just noticed that it should be "the vector which when added to the accelerometer measurements" (I think I got it mixed up because I needed to subtract the gravity vector when I was coming up with simulated accelometer measurements).

Reply:
Yeah ok that makes sense, I was just trying to avoid "true acceleration" being ambiguous between the acceleration relative to the world frame and acceleration relative to some inertial frame (in that the raw acceleration the imu measures is its "true acceleration" in the GR sense)